src/ptunnel.c

```diff
@@ -558,10 +557,23 @@ void* pt_proxy(void *args) {
 			if (cur->state == kProxy_start) {
 				pt_log(kLog_verbose, "Sending proxy request.\n");
 				cur->last_ack = time_as_double();
-				queue_packet(fwd_sock, cur->pkt_type, 0, 0, cur->id_no, cur->id_no,
+				uint16_t *extended_options = 0;
+				size_t extended_options_size = 0;
+				if (opts.window_size || opts.ack_interval || opts.resend_interval || opts.payload_size) {
+					extended_options = calloc(4, sizeof(uint16_t));
```

Comment:
Why not stack allocated?

Reply:
Extended options are optional, so the intent is to pass a null pointer to queue_packet to indicate an empty data section. Since queue_packet checks the data length also, the client might as well send a variable length data section.